### PR TITLE
Validate arguments to MyAvatar.addAnimationStateHandler()

### PR DESCRIPTION
**Pull request by hyperlogic**
Also, validate arguments to MyAvatar.removeAnimationStateHandler() and the return result from the user provided callback function.

### CODE REVIEW

**Comment by hyperlogic** [comment]
https://app.asana.com/0/45663370065758/112887140597782

**Comment by hyperlogic** [comment]
To test: run the following script command:

    MyAvatar.addAnimationStateHandler("inAirAlpha",0);

in this case the first argument is of the incorrect type, it should be a function not a string.  On the release build this will result in a crash, in this PR it should not crash and print a warning message int the console:

    [WARNING] Rig::addAnimationStateHandler invalid arguments, expected (function, string[])

**Comment by hifi-gustavo** [comment]
Build console output for commit: 253e4cbb73ca62045269e1ad566f5c4f23fa69ca available at: https://gist.github.com/43b7d8bd8d36d830cf57d5c526354dad<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7670.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7670.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 253e4cbb73ca62045269e1ad566f5c4f23fa69ca available at: https://gist.github.com/43b7d8bd8d36d830cf57d5c526354dad<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7670.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7670.dmg

File: libraries/animation/src/Rig.cpp [hunk 1]
```diff
@@ -783,25 +783,55 @@ void Rig::computeMotionAnimationState(float deltaTime, const glm::vec3& worldPos
     _lastVelocity = workingVelocity;
 }
 
+static bool isListOfStrings(const QScriptValue& arg) {
+    if (!arg.isArray())
```
Comment (ZappoMan) [inline libraries/animation/src/Rig.cpp]: coding standard: braces

File: libraries/animation/src/Rig.cpp [hunk 2]
```diff
@@ -783,25 +783,55 @@ void Rig::computeMotionAnimationState(float deltaTime, const glm::vec3& worldPos
...
+        return false;
+
+    auto lengthProperty = arg.property("length");
+    if (!lengthProperty.isNumber())
```
Comment (ZappoMan) [inline libraries/animation/src/Rig.cpp]: coding standard: braces

File: libraries/animation/src/Rig.cpp [hunk 3]
```diff
@@ -783,25 +783,55 @@ void Rig::computeMotionAnimationState(float deltaTime, const glm::vec3& worldPos
     _lastVelocity = workingVelocity;
 }
 
+static bool isListOfStrings(const QScriptValue& arg) {
```
Comment (ZappoMan) [inline libraries/animation/src/Rig.cpp]: this should maybe be in the shared library?

**Comment by Menithal** [comment]
Correctly reports in Log that invalid arguments and no longer crashes the client.

No logging occurs when correct function, list is submitted.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 38418d016915624f7c8c247f5ea75689212e0f1d available at: https://gist.github.com/43b7d8bd8d36d830cf57d5c526354dad<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7670.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7670.dmg

**Pull request merged**